core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2ServerConfigProvider.java

```diff
@@ -31,7 +31,6 @@ record OAuth2ServerConfig(Optional<String> accessTokenIssuer, URI authUrl, URI t
             requireNonNull(tokenUrl, "tokenUrl is null");
             requireNonNull(jwksUrl, "jwksUrl is null");
             requireNonNull(userinfoUrl, "userinfoUrl is null");
-            requireNonNull(endSessionUrl, "endSessionUrl is null");
```

Comment:
Why do we need to remove the `rnn` check here - It can be `Optional#empty` right ?